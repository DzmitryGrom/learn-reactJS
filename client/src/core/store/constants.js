export const SEARCH_MOVIES_TEXT = 'SEARCH_MOVIES_TEXT';
export const SEARCH_MOVIES = 'SEARCH_MOVIES';
export const SORT_MOVIES_BY_RELEASE = 'SEARCH_MOVIES_BY_RELEASE';
export const SORT_MOVIES_BY_RATING = 'SORT_MOVIES_BY_RATING';
export const SORT_BY_RELEASE = 'release_date';
export const SORT_BY_RATING = 'vote_average';
export const SEARCH_TEXT = 'SEARCH_TEXT';
export const SET_MOVIES = 'SET_MOVIES';
