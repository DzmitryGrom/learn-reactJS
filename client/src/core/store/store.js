import { createStore, compose } from 'redux';

import { persistStore, persistReducer } from 'redux-persist';
import storage from 'redux-persist/lib/storage';

import rootReducer from './root-reducer';

const persistConfig = {
    key: 'root',
    storage,
},
    reducer = persistReducer(persistConfig, rootReducer),
    composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

export default () => {
    const store = createStore(reducer, composeEnhancers()),
        persistor = persistStore(store);
    return { store, persistor };
}