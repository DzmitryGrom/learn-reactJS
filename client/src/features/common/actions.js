import { SET_MOVIES } from '../../core/store/constants';

export const setMovies = movies => ({
  type: SET_MOVIES,
  payload: movies
})