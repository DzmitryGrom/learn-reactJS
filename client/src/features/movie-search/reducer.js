import { SET_MOVIES } from '../../core/store/constants';

const initialState = [];

export default function reducer(state = initialState, action) {
    switch (action.type) {
        case SET_MOVIES:
            return [...action.payload]

        default:
            return state
    }
}