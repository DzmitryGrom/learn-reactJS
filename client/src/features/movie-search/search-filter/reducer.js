import * as type from '../../../core/store/constants';

const initialState = {
  sortBy: '',
}

export default function reducer(state = initialState, action) {
  switch (action.type) {
    case type.SORT_MOVIES_BY_RELEASE:
      return {
        ...state, sortBy: action.payload
      }
    case type.SORT_MOVIES_BY_RATING:
      return {
        ...state, sortBy: action.payload
      }
    default:
      return state
  }
}