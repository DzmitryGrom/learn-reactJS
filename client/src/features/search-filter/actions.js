import * as type from '../../core/store/constants';

export const searchByTitle = () => ({
    type: type.SEARCH_MOVIES_BY_TITLE,
    payload: type.SEARCH_BY_TITLE
})
export const searchByGenre = () => ({
    type: type.SEARCH_MOVIES_BY_GENRE,
    payload: type.SEARCH_BY_GENRE
})
export const sortByRelease = () => ({
    type: type.SORT_MOVIES_BY_RELEASE,
    payload: type.SORT_BY_RELEASE
})
export const sortByRating = () => ({
    type: type.SORT_MOVIES_BY_RATING,
    payload: type.SORT_BY_RATING
})
export const setSearchText = text => ({
    type: type.SEARCH_TEXT,
    payload: text
})