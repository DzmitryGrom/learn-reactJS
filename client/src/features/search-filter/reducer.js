import * as type from '../../core/store/constants';

const initialState = {
    searchBy: 'title',
    sortBy: '',
    text: 'vote_average'
}

export default function reducer(state = initialState, action) {
    switch (action.type) {
        case type.SEARCH_MOVIES:
            return {
                ...state, 
                searchBy: action.payload
            }
        case type.SORT_MOVIES_BY_RELEASE:
            return {
                ...state, 
                sortBy: action.payload
            }
        case type.SORT_MOVIES_BY_RATING:
            return {
                ...state, 
                sortBy: action.payload
            }

        case type.SEARCH_TEXT:
            return {
                ...state, 
                text: action.payload
            }


        default:
            return state
    }
}