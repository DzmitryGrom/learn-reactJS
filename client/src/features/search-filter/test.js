import React from 'react';
import { mount } from 'enzyme';
import Element from './index';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import * as action from './actions';
import reducer from './reducer';
import  * as type from '../../core/store/constants';

const mockStore = configureStore(),
  mockStoreInitialized = mockStore({ 
      films: [ 
          {
              "data": [
                {
                  "id": 0,
                  "title": "string",
                  "tagline": "string",
                  "vote_average": 0,
                  "vote_count": 0,
                  "release_date": "string",
                  "poster_path": "string",
                  "overview": "string",
                  "budget": 0,
                  "revenue": 0,
                  "runtime": 0,
                  "genres": [
                    "string"
                  ]
                }
              ],
              "total": 0,
              "offset": 0,
              "limit": 0
            }
      ]
  }); 

describe('tests search-filter', () => {
    it('render  component with empty redux', () => {
        const wrapper = mount(<Provider store={mockStoreInitialized}><Element/></Provider>),
            component = wrapper.find(Element);
        expect(component).toMatchSnapshot();
        expect(component.find('.filter__buttonsLabel').text()).toBe('search by');
        expect(component.find('.btn')).toHaveLength(1);
        expect(component.find('[type="radio"]')).toHaveLength(4);
        expect(wrapper.props().store.getState().filter).toBeFalsy();
    });
    it('test action searchByTitle', () => {
            const expectedAction = {
                type: type.SEARCH_MOVIES_BY_TITLE,
                payload: type.SEARCH_BY_TITLE
            },
            badExpectedAction = {
                type: type.SEARCH_MOVIES_BY_TITLE + 'some',
                payload: type.SEARCH_BY_TITLE + 'some'
            };
        expect(action.searchByTitle()).toEqual(expectedAction);
        expect(action.searchByTitle()).not.toEqual(badExpectedAction);
    });
    it('test action searchByGenre', () => {
            const expectedAction = {
                type: type.SEARCH_MOVIES_BY_GENRE,
                payload: type.SEARCH_BY_GENRE
            },
            badExpectedAction = {
                type: type.SEARCH_MOVIES_BY_GENRE + 'some',
                payload: type.SEARCH_BY_TITLE + 'some'
            };
        expect(action.searchByGenre()).toEqual(expectedAction);
        expect(action.searchByGenre()).not.toEqual(badExpectedAction);
    });
    it('test action sortByRelease', () => {
            const expectedAction = {
                type: type.SORT_MOVIES_BY_RELEASE,
                payload: type.SORT_BY_RELEASE
            },
            badExpectedAction = {
                type: type.SORT_MOVIES_BY_RELEASE + 'some',
                payload: type.SORT_BY_RELEASE + 'some'
            };
        expect(action.sortByRelease()).toEqual(expectedAction);
        expect(action.sortByRelease()).not.toEqual(badExpectedAction);
    });
    it('test action sortByRating', () => {
        const expectedAction = {
            type: type.SORT_MOVIES_BY_RATING,
            payload: type.SORT_BY_RATING
        },
        badExpectedAction = {
            type: type.SORT_MOVIES_BY_RATING  + 'some',
            payload: type.SORT_BY_RATING + 'some'
        };
        expect(action.sortByRating()).toEqual(expectedAction);
        expect(action.sortByRating()).not.toEqual(badExpectedAction);
    });
    it('test action setSearchText', () => {
        const expectedAction = {
            type: type.SEARCH_TEXT,
            payload: 'Some'
        },
        badExpectedAction = {
            type: type.SEARCH_TEXT + 'some',
            payload: 'Some'
        
        };
        expect(action.setSearchText('Some')).toEqual(expectedAction);
        expect(action.setSearchText('Some')).not.toEqual(badExpectedAction);
    });
    it('tests reducers', () => {
        expect(
            reducer([], {
                type: type.SEARCH_MOVIES_BY_TITLE,
                payload: type.SEARCH_BY_TITLE
            })
        ).toEqual({ "searchBy": "title" });

        expect(
            reducer([], {
                type: type.SEARCH_MOVIES_BY_GENRE,
                payload: type.SEARCH_BY_GENRE
            })
        ).toEqual({ "searchBy": "genres" });

        expect(
            reducer([], {
                type: type.SORT_MOVIES_BY_RATING,
                payload: type.SORT_BY_RATING
            })
        ).toEqual({ "sortBy": "vote_average" });

        expect(
            reducer([], {
                type: type.SORT_MOVIES_BY_RELEASE,
                payload: type.SORT_BY_RELEASE
            })
        ).toEqual({ "sortBy": "release_date" });

        expect(
            reducer([], {
                type: type.SEARCH_TEXT,
                payload: 'some'
            })
        ).toEqual({ "text": "some" });
    });
})