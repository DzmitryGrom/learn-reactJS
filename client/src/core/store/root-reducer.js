import { combineReducers } from 'redux';
import {default as films} from '../../features/movie-search/reducer';
import {default as filter} from '../../features/movie-search/search-filter/reducer';
export default combineReducers({ films, filter })