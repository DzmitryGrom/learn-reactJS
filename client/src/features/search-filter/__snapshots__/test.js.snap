// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests search-filter render  component with empty redux 1`] = `
<Connect(FilterContainer)>
  <FilterContainer
    searchByGenre={[Function]}
    searchByTitle={[Function]}
    setSearchText={[Function]}
    sortByRating={[Function]}
    sortByRelease={[Function]}
  >
    <FilterComponent
      getInputRef={[Function]}
      onButtonClick={[Function]}
      onButtonClickGenre={[Function]}
      onButtonClickRating={[Function]}
      onButtonClickRelease={[Function]}
      onButtonClickTitle={[Function]}
    >
      <div
        className="filter"
      >
        <div
          className="filter__top"
        >
          <span
            className="filter__label"
          >
            Find your movie
          </span>
          <input
            className="filter__input"
            placeholder="Type your value"
            type="text"
          />
          <div
            className="filter__buttons"
          >
            <span
              className="filter__buttonsLabel"
            >
              search by
            </span>
            <input
              id="btnTitle"
              name="search"
              onClick={[Function]}
              type="radio"
            />
            <label
              className="filter__btn"
              htmlFor="btnTitle"
            >
              title
            </label>
            <input
              id="btnGenre"
              name="search"
              onClick={[Function]}
              type="radio"
            />
            <label
              className="filter__btn"
              htmlFor="btnGenre"
            >
              genre
            </label>
          </div>
          <Button
            modifier="pink"
            onButtonClick={[Function]}
            selector="btnSearch"
            size="big"
            text="search"
          >
            <input
              className="btn btn_pink btn_big"
              id="btnSearch"
              onClick={[Function]}
              type="button"
              value="search"
            />
          </Button>
        </div>
        <div
          className="filter__btns"
        >
          <span
            className="filter__value"
            id="btnSortBy"
          >
            Sort by
          </span>
          <input
            id="btnRelease"
            name="sort"
            onClick={[Function]}
            type="radio"
          />
          <label
            className="filter__value"
            htmlFor="btnRelease"
          >
            release date
          </label>
          <input
            id="btnRating"
            name="sort"
            onClick={[Function]}
            type="radio"
          />
          <label
            className="filter__value"
            htmlFor="btnRating"
          >
            rating
          </label>
        </div>
      </div>
    </FilterComponent>
  </FilterContainer>
</Connect(FilterContainer)>
`;
